docs/changelog/720.bugfix.rst

```diff
@@ -0,0 +1 @@
+do not allow setting the ``TOXENV`` or the ``-e`` flag to override the listed default environment variables, they still show up under extra if non defined target - by :user:`gaborbernat`
```

Comment:
maybe mention that this is about `tox -l`, took me  a while of reading the code to figure that out :D

Reply:
done